app/src/editor/canvas/tests/mocks.js

```diff
@@ -1,18 +1,33 @@
 import uuid from 'uuid'
 
-function remapIds(items) {
-    return items.map((item) => ({
-        ...item,
-        id: uuid(),
-    }))
+function remapIds(items, index = {}) {
+    return items.map((item) => {
+        if (!index[item.id]) {
+            index[item.id] = uuid()
+        }
+
+        return {
+            ...item,
+            id: index[item.id],
+        }
+    })
+}
+
+function remapModule(data, index = {}) {
+    const newData = { ...data }
+    if (newData.inputs) { newData.inputs = remapIds(newData.inputs, index) }
+    if (newData.outputs) { newData.outputs = remapIds(newData.outputs, index) }
+    if (newData.params) { newData.params = remapIds(newData.params, index) }
+    return newData
 }
 
 function createMockModuleFactory(data) {
     return () => {
-        const newData = { ...data }
-        if (newData.inputs) { newData.inputs = remapIds(newData.inputs) }
-        if (newData.outputs) { newData.outputs = remapIds(newData.outputs) }
-        if (newData.params) { newData.params = remapIds(newData.params) }
+        const index = {}
+        const newData = remapModule(data, index)
```

Comment:
Applying `remapModule` on the module itself probably isn't correct, the module's `id` shouldn't change as `id` refers to the module's "type" e.g. `Table` modules will always have the same `id`.  It's not ideal but the module's `hash` is the unique identifier, but also note it can't be a uuid as the server wants hash to be a 32bit number.

Modules already get a new hash generated when they are added to the canvas in `addModule` so likely nothing needs to be done to make this work in test.

Reply:
Ok, the `id`/`hash` is still a bit confusing to me. I was trying to make it work so that the ports and output lifted from the subcanvas elements would stay the same. Although that's just testing the mock content and not the code...